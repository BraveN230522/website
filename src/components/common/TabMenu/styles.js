import styled from '@emotion/styled';

const ListMenu = styled.ul`
  list-style-type: none;
  padding: 0;

  ${({ theme }) => theme.media.phone`
    li a {
      margin-right: 16px;
    }
    &:first-of-type a {
      margin-left: 10px;
    }
  `}
`;

const ItemMenu = styled.li`
  display: inline-block;

  &:last-of-type a {
    margin-right: 0;
  }
`;

const LinkMenu = styled.a`
  color: ${({ theme }) => theme.colors.white};
  cursor: pointer;
  display: inline-block;
  font-family: ${({ theme }) => theme.fonts.Inter};
  font-size: 21px;
  font-weight: 400;
  line-height: 28px;
  letter-spacing: -0.1465116px;
  margin-right: 40px;
  padding-bottom: 2px;
  text-decoration: none;
  border-bottom: solid 2px transparent;
  transition: all 0.3s ease-in;

  &:hover {
    text-decoration: none;
    border-bottom: solid 2px ${({ theme, activeLink }) => (activeLink ? theme.colors.swiamBlue : theme.colors.white)};
  }

  li:last-child > & {
    padding-right: 0;
  }

  ${({ theme }) =>
    theme.media.phone(
      `
      font-size: 18px;
      line-height: 21px;
      letter-spacing: -0.1116279px;
      opacity: 0.6;

      &:hover {
        opacity: 1;
      }
  `
    )};

  ${({ theme }) =>
    theme.generateBreakpoint(
      466,
      `
    font-size: 16px;
  `
    )};

  ${({ theme }) =>
    theme.generateBreakpoint(
      360,
      `
      font-size: 14px;
`
    )};

  & {
    opacity: ${({ activeLink }) => activeLink && 1};
    ${({ theme, activeLink }) =>
      activeLink &&
      `
      border-bottom: 2px solid ${theme.colors.swiamBlue};
      font-weight: 600;
      `};
  }
`;

const ContentMenu = styled('div')`
  display: flex;
  flex-wrap: wrap;

  ${({ theme }) => theme.generateBreakpoint(1252, `justify-content: center;`)}
`;

export { ListMenu, ItemMenu, LinkMenu, ContentMenu };
