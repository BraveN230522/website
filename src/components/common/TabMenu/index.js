import React from 'react';

import useSelection from '@utils/hooks/useSelection';
import More from '@components/common/More';
import { ListMenu, ItemMenu, LinkMenu, ContentMenu } from './styles';

const TabMenu = ({ items, className }) => {
  const [activeItem, activeIndex, changeActiveItem] = useSelection({ items, selectedIndex: 0 });

  const handleTabItemClick = index => () => changeActiveItem(index);

  return (
    <>
      <ListMenu className={className}>
        {items &&
          items.map((item, index) => (
            <ItemMenu key={index}>
              <LinkMenu activeLink={activeIndex === index} onClick={handleTabItemClick(index)}>
                {item.name}
              </LinkMenu>
            </ItemMenu>
          ))}
      </ListMenu>

      <ContentMenu>{activeItem.renderContent()}</ContentMenu>
    </>
  );
};

export default TabMenu;
