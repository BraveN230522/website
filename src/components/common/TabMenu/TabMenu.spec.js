import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import withGlobals from '@swiamTestUtils/withGlobals';

import TabMenu from './index';

const menu = [
  { name: 'Popular', renderContent: () => 'Popular content' },
  { name: 'Events', renderContent: () => 'Events content' },
  { name: 'Cities', renderContent: () => 'Cities content' },
  { name: 'Hotels', renderContent: () => 'Hotels content' },
  { name: 'Experiences', renderContent: () => 'Experiences content' },
];

describe('<TabMenu />', () => {
  it('Renders TabMenupage unchanged', () => {
    const { container } = render(<TabMenu items={menu} />, {
      wrapper: withGlobals,
    });
    expect(container).toMatchSnapshot();
  });

  it('Changes the active item when an item is clicked', () => {
    const { name, renderContent } = menu[1];
    const { getByText, container } = render(<TabMenu items={menu} />, {
      wrapper: withGlobals,
    });

    fireEvent.click(getByText(name));

    screen.getByText(renderContent());

    expect(container).toHaveTextContent(renderContent());
  });

  it('Should keep the same content when an active item is clicked', () => {
    const { name, renderContent } = menu[0];
    const { getByText, container } = render(<TabMenu items={menu} />, {
      wrapper: withGlobals,
    });

    fireEvent.click(getByText(name));

    screen.getByText(renderContent());

    expect(container).toHaveTextContent(renderContent());
  });
});
