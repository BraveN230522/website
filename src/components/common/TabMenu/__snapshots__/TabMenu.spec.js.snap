// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabMenu /> Renders TabMenupage unchanged 1`] = `
.emotion-20 {
  list-style-type: none;
  padding: 0;
}

@media (max-width:48em) {
  .emotion-20 li a {
    margin-right: 16px;
  }

  .emotion-20:first-of-type a {
    margin-left: 10px;
  }
}

.emotion-2 {
  display: inline-block;
}

.emotion-2:last-of-type a {
  margin-right: 0;
}

.emotion-0 {
  color: #ffffff;
  cursor: pointer;
  display: inline-block;
  font-family: Inter,sans-serif;
  font-size: 21px;
  font-weight: 400;
  line-height: 28px;
  -webkit-letter-spacing: -0.1465116px;
  -moz-letter-spacing: -0.1465116px;
  -ms-letter-spacing: -0.1465116px;
  letter-spacing: -0.1465116px;
  margin-right: 40px;
  padding-bottom: 2px;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-bottom: solid 2px transparent;
  -webkit-transition: all 0.3s ease-in;
  transition: all 0.3s ease-in;
}

.emotion-0:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  border-bottom: solid 2px #2891ff;
}

li:last-child > .emotion-0 {
  padding-right: 0;
}

@media (max-width:48em) {
  .emotion-0 {
    font-size: 18px;
    line-height: 21px;
    -webkit-letter-spacing: -0.1116279px;
    -moz-letter-spacing: -0.1116279px;
    -ms-letter-spacing: -0.1116279px;
    letter-spacing: -0.1116279px;
    opacity: 0.6;
  }

  .emotion-0:hover {
    opacity: 1;
  }
}

@media (max-width:29.125em) {
  .emotion-0 {
    font-size: 16px;
  }
}

@media (max-width:22.5em) {
  .emotion-0 {
    font-size: 14px;
  }
}

.emotion-0 {
  opacity: 1;
  border-bottom: 2px solid #2891ff;
  font-weight: 600;
}

.emotion-4 {
  color: #ffffff;
  cursor: pointer;
  display: inline-block;
  font-family: Inter,sans-serif;
  font-size: 21px;
  font-weight: 400;
  line-height: 28px;
  -webkit-letter-spacing: -0.1465116px;
  -moz-letter-spacing: -0.1465116px;
  -ms-letter-spacing: -0.1465116px;
  letter-spacing: -0.1465116px;
  margin-right: 40px;
  padding-bottom: 2px;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-bottom: solid 2px transparent;
  -webkit-transition: all 0.3s ease-in;
  transition: all 0.3s ease-in;
}

.emotion-4:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  border-bottom: solid 2px #ffffff;
}

li:last-child > .emotion-4 {
  padding-right: 0;
}

@media (max-width:48em) {
  .emotion-4 {
    font-size: 18px;
    line-height: 21px;
    -webkit-letter-spacing: -0.1116279px;
    -moz-letter-spacing: -0.1116279px;
    -ms-letter-spacing: -0.1116279px;
    letter-spacing: -0.1116279px;
    opacity: 0.6;
  }

  .emotion-4:hover {
    opacity: 1;
  }
}

@media (max-width:29.125em) {
  .emotion-4 {
    font-size: 16px;
  }
}

@media (max-width:22.5em) {
  .emotion-4 {
    font-size: 14px;
  }
}

.emotion-22 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

@media (max-width:78.25em) {
  .emotion-22 {
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

<div>
  <ul
    class="emotion-20 emotion-21"
  >
    <li
      class="emotion-2 emotion-3"
    >
      <a
        class="emotion-0 emotion-1"
      >
        Popular
      </a>
    </li>
    <li
      class="emotion-2 emotion-3"
    >
      <a
        class="emotion-4 emotion-1"
      >
        Events
      </a>
    </li>
    <li
      class="emotion-2 emotion-3"
    >
      <a
        class="emotion-4 emotion-1"
      >
        Cities
      </a>
    </li>
    <li
      class="emotion-2 emotion-3"
    >
      <a
        class="emotion-4 emotion-1"
      >
        Hotels
      </a>
    </li>
    <li
      class="emotion-2 emotion-3"
    >
      <a
        class="emotion-4 emotion-1"
      >
        Experiences
      </a>
    </li>
  </ul>
  <div
    class="emotion-22 emotion-23"
  >
    Popular content
  </div>
</div>
`;
